javascript/game.spec.js

```diff
@@ -162,4 +162,51 @@ describe('all tests!', () => {
     })
   })
 
+  describe("questions", function () {
+
+
+
+    it('When asking a question deck of questions remains the same length', () => {
+      const game = new Game;
+
+      const popQuestions = game.getQuestionsArray()[0];
+
+      for (let i = 0; i < 50 ; i++) {
+
+      }
+      game.askQuestionFromCategory('Pop');
+      expect(popQuestions.questions.length).toEqual(50);
+
+    });
+
+    it('should keep track of the current index of questions and the index should increment expectedly', () => {
+      const game = new Game;
+
+      const numberOfQuestionsAsked = 10;
+      const popObj = game.getQuestionsArray()[0];
+
+      for (let i = 0; i < numberOfQuestionsAsked; i++) {
+        game.askQuestion('Pop');
+      }
+
+      expect(popObj.index).toEqual(10)
+      // TODO - expect the index to have moved to numberOfQuestionsAsked.
+      // expect(game.getPopQuestions().currentQuestion).toEqual(numberOfQuestionsAsked);
+    });
+
+    it("more questions asked than there are in deck, should go back to the beginning of the deck", () => {
+      const game = new Game;
+
+      const numberOfQuestionAsked = 51;
```

Comment:
Magic number!
What is a programmatic way of determining the number of questions to ask?